File: 2024/day01/part2.py

def sanitize(data: str) -> tuple[list[int], list[int]]:
    left = []
    right = []

    for line in data.split("\n")[:-1]:
        [litem, ritem] = line.split()
        left.append(int(litem))
        right.append(int(ritem))
    
    return left, right

def solve(data: str) -> str:
    left, right = sanitize(data)
    
    result = 0
    for item in left:
        result += item * right.count(item)
    
    return str(result)


filecontent = ""
with open("./2024/day01/input.txt") as file:
    filecontent = file.read()

result = solve(filecontent)
print(result)

with open("./2024/day01/result2.txt", "w") as file:
    file.write(str(result))

File: 2024/day03/part2.exs
defmodule Part2 do
  @spec solve(String.t()) :: String.t()
  def solve(data), do:
  Regex.scan(~r/mul\((?<first>\d+),(?<second>\d+)\)/, sanitize(data))
    |> Enum.map(fn [_, first, second] -> [String.to_integer(first), String.to_integer(second)] end)
    |> Enum.map(fn [first, second] -> first*second end)
    |> Enum.sum()
    |> Integer.to_string()

  def sanitize(data), do:
    Regex.replace(~r/don't\(\).*$/,
      Regex.replace(~r/don't\(\).*do\(\)/U,
        data,
      ""),
    "")
end

data = File.read!("./2024/day03/input.txt")
result = Part2.solve(data)
IO.puts(result)
File.write!("./2024/day03/result2.txt", result)
